test/functional/lua/version_spec.lua

```diff
@@ -100,17 +100,31 @@ describe('version', function()
       end)
 
       it('[to] not in range ' .. input .. ' to:' .. tostring(range.to), function()
-        if range.to then
+        if range.to and range.to ~= range.from then
           assert(not (range.to < range.to))
           assert(not range:has(range.to))
         end
       end)
+
+      it('[to] in range ' .. input .. ' to:' .. tostring(range.to), function()
+        if range.to and range.to == range.from then
+          assert(range:has(range.to))
+        end
+      end)
     end
 
     it('handles prerelease', function()
-      assert(not vim.version.range('1.2.3'):has('1.2.3-alpha'))
-      assert(vim.version.range('1.2.3-alpha'):has('1.2.3-alpha'))
-      assert(not vim.version.range('1.2.3-alpha'):has('1.2.3-beta'))
+      assert(not vim.version.range('1.2.3'):has('1.2.3-alpha'), '1.2.3 should not have 1.2.3-alpha')
```

Comment:
don't need the extra messages. the code is already quite readable. though it was wrong before :)
```suggestion
      assert(not vim.version.range('1.2.3'):has('1.2.3-alpha'))
```